src/new/wallet.js

```diff
@@ -0,0 +1,441 @@
+ * - CONNECTING: When it is connecting to the server.
+ * - SYNCING: When it has connected and is syncing the transaction history.
+ * - READY: When it is ready to be used.
+ *
+ * You can subscribe for the following events:
+ * - state: Fired when the state of the Wallet changes.
+ * - new-tx: Fired when a new tx arrives.
+ * - update-tx: Fired when a known tx is updated. Usually, it happens when one of its outputs is spent.
+ * - more-addresses-loaded: Fired when loading the history of transactions. It is fired multiple times,
+ *                          one for each request sent to the server.
+ **/
+class Wallet extends EventEmitter {
+  constructor({ network, server, seed, tokenUid }) {
+    super();
+
+    networkInstance.setNetwork(network);
+    this.network = network;
+
+    this.state = Wallet.CLOSED;
+    this.serverInfo = null;
+
+    this.onConnectionChange = this.onConnectionChange.bind(this);
+    this.handleWebsocketMsg = this.handleWebsocketMsg.bind(this);
+    this.onAddressesLoaded = this.onAddressesLoaded.bind(this);
+
+    this.seed = seed;
+    this.server = server;
+    if (!this.server) {
+      // We set the default server of the lib
+      this.server = DEFAULT_SERVER;
+
+    }
+
+    // tokenUid is optional so we can get the token of the wallet
+    this.token = null;
+    this.tokenUid = tokenUid;
+
+
+    this.passphrase = '';
+    // XXX Update it so we don't have fixed pin/password
+    this.pinCode = '123';
+    this.password = '123';
+  }
+
+  /**
+   * Called when loading the history of transactions.
+   * It is called every HTTP Request to get the history of a set of addresses.
+   * Usually, this is called multiple times.
+   * The `historyTransactions` contains all transactions, including the ones from a previous call to this method.
+   *
+   * @param {Object} result {historyTransactions, allTokens, newSharedAddress, newSharedIndex, addressesFound}
+   **/
+  onAddressesLoaded(result) {
+    this.emit('more-addresses-loaded', result);
+  }
+
+  /**
+   * Called when the connection to the websocket changes.
+   * It is also called if the network is down.
+   **/
+  onConnectionChange(value) {
+    if (value) {
+      this.setState(Wallet.SYNCING);
+      wallet.loadAddressHistory(0, GAP_LIMIT).then(() => {
+        this.setState(Wallet.READY);
+      }).catch((error) => {
+        throw error;
+      })
+    } else {
+      // CONNECTING or CLOSED?
+      this.serverInfo = null;
+      this.setState(Wallet.CONNECTING);
+    }
+  }
+
+  getAddressToUse() {
+    return wallet.getAddressToUse();
+  }
+
+  getCurrentAddress() {
+    return wallet.getCurrentAddress();
+  }
+
+  /**
+   * Called when a new message arrives from websocket.
+   **/
+  handleWebsocketMsg(wsData) {
+    if (wsData.type === 'wallet:address_history') {
+      this.onNewTx(wsData);
+    }
+  }
+
+  reloadData() {
+    // TODO Reload data?
+    console.log('reloadData');
+  }
+
+  getBalance(tokenUid) {
+    if (!tokenUid && !this.token) {
```

Comment:
I don't like it. Would we need to pass `"HTR_TOKEN"` on every call?

Reply:
No. If you start the wallet and have `tokenUid` as '00' you don't need to pass any parameter in this method. You only need to pass parameter if you want something different from the instance.

The behaviour you said you don't like is exactly what was before.

Reply:
As we will discuss the best interface later, I think we can move forward.